src/backend/metal/src/device.rs

```diff
@@ -1575,7 +1575,10 @@ impl hal::Device<Backend> for Device {
         }
 
         if let Some(fun) = info.comparison {
-            descriptor.set_compare_function(conv::map_compare_function(fun));
+            if (caps.os_is_mac && caps.has_version_at_least(10, 11)) ||
+                (!caps.os_is_mac && caps.has_version_at_least(9, 0)) {
+                descriptor.set_compare_function(conv::map_compare_function(fun));
```

Comment:
We should return an error otherwise. If the user requests a comparison function, we can't just ignore it.

Reply:
But there is not suitable error of `AllocationError` for the case.

Reply:
If we can't return an error, we should panic. It's not like the user is going to get anything meaningful anyway if they ask for a comparison sampler but don't get it.

Reply:
Should we add a bool field in `PrivateCapabilities` to indicate that whether `set_compare_function` works on the current platform?